test/functional.js

```diff
@@ -583,6 +583,50 @@ module.exports = {
             });
         });
 
+        it('When configured, TypeScript is compiled!', (done) => {
+            const config = createWebpackConfig('www/build', 'dev');
+            config.setPublicPath('/build');
+            config.addEntry('main', ['./js/render.ts', './js/index.ts']);
+            const testCallback = () => {};
+            config.configureTypeScript(testCallback);
+
+            testSetup.runWebpack(config, (webpackAssert) => {
+                // check that ts-loader transformed the ts file
+                webpackAssert.assertOutputFileContains(
+                    'main.js',
+                    'document.getElementById(\'wrapper\').innerHTML = "<h1> Hello World!</h1>";'
+                );
+
+                done();
+            });
+        });
+
+        it('When enabled, react JSX and TypeScript work nice together!', (done) => {
+            const config = createWebpackConfig('www/build', 'dev');
+            config.setPublicPath('/build');
+            config.addEntry('main', ['./js/CoolReactComponent.jsx', './js/render2.tsx']);
+            config.enableReactPreset();
+            config.configureTypeScript(function(tsConfig) {
+                tsConfig.compilerOptions = { 'jsx': 'preserve' };
```

Comment:
It is a TypeScript compilation issue. Remove it and you get compilation errors about not having configure this option. This only happen when compiling `.tsx` files. I did not add a check/validation for that when configuring loader because:
* Not sure where to add it.
* Difficult to know before hand which kind of files will be parsed ? Not sure here though.
* The compilation error is pretty clear. Developers should configure the option, that's it.

Take a look to [this example configuration](https://github.com/TypeStrong/ts-loader/blob/master/examples/react-babel-karma-gulp-happypack/tsconfig.json) I just tooked from there. I am not an expert so most of test/functional/usage I have took them from  `ts-loader` project itself